Curtains/Curtains.Infrastructure/Repositories/AccessoriesRepository.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Curtains.Domain.Models;
+using Curtains.Infrastructure.Database;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Curtains.Infrastructure.Repositories
+{
+    public class AccessoriesRepository : IRepository<AccessoriesModel>
+    {
+        private readonly ILogger _logger;
+        private readonly CurtainsDbContext _curtainsContext;
+
+        public AccessoriesRepository(CurtainsDbContext curtainsContext, ILogger logger)
+        {
+            _curtainsContext = curtainsContext;
+            _logger = logger;
+        }
+
+        public IEnumerable<AccessoriesModel> GetAll()
+        {
+            return _curtainsContext.Accessories.AsEnumerable();
+        }
+
+        public async Task<AccessoriesModel> GetById(int Id)
+        {
+            var result = await _curtainsContext.Accessories.FirstOrDefaultAsync(c => c.Id == Id);
+
+            if (result != null)
```

Comment:
How can this if be useful for us?  🤔

Reply:
Please add logger in this validation instead of return null